test/total.jl

```diff
@@ -73,28 +73,51 @@ end
 
 @testset "Operator Algebra" begin
     N2 = N*N
+
     A = rand(N,N)
-    B = rand(N,N)
+    # Introduce update function for B
+    B = MatrixOperator(zeros(N,N); update_func=(A, u, p, t) -> (A .= p))
     C = rand(N,N)
-    D = rand(N,N)
+    # Introduce update function for D dependent on kwarg "matrix"
+    D = MatrixOperator(zeros(N,N); update_func=(A, u, p, t; matrix) -> (A .= p*t*matrix), 
+                       accepted_kwargs=(:matrix,))
 
     u = rand(N2,K)
+    p = rand()
+    t = rand()
+    matrix = rand(N, N)
+    diag = rand(N2)
     α = rand()
     β = rand()
 
     T1 = ⊗(A, B)
     T2 = ⊗(C, D)
 
-    D1  = DiagonalOperator(rand(N2))
-    D2  = DiagonalOperator(rand(N2))
+    # Introduce update function for D1
+    D1  = DiagonalOperator(p * ones(N2); update_func=(A, u, p, t) -> (A .= p))
```

Comment:
can you rename dummy variable `A` to `d` to indicate that it is a vector. Uppercase notation is usually for matrices.

Reply:
Can you also initialize the operator with zeros like `DiagonalOperator(zeros(N); update_func...)` (currently you are initializing with the right answer). This is so we can concretely see `update_coeffs` in action